utils/ccloud_library.sh

```diff
@@ -1055,8 +1056,11 @@ function ccloud::destroy_ccloud_stack() {
 
   echo "Destroying Confluent Cloud stack associated to service account id $SERVICE_ACCOUNT_ID"
 
+  local cluster_id=$(confluent kafka cluster list -o json | jq -r 'map(select(.name == "'"$CLUSTER_NAME"'")) | .[].id')
+  confluent kafka cluster use $cluster_id 2>/dev/null
```

Comment:
Given this statement, is it possible that the other `--cluster $CLUSTER_ID` are extraneous and can be removed?

Reply:
I decided both `--cluster` instances were safest and correct to retain, albeit unnecessary following successful `confluent kafka cluster use`.  Ideally we don't rely on this context for any commands - ideally all the commands are independent of context by fully specifying context via args.

But no, they aren't necessary, if removed I expect the bug would remain fixed.

Reply:
For simplicity (and therefore maintenance0, it may be better to keep this PR to just these 2 lines here

```
  local cluster_id=$(confluent kafka cluster list -o json | jq -r 'map(select(.name == "'"$CLUSTER_NAME"'")) | .[].id')
   confluent kafka cluster use $cluster_id 2>/dev/null
```

And remove the redundancy of the other lines in this PR (`--cluster $CLUSTER_ID`)

Reply:
@ybyzek how would you feel about instead retaining the `--cluster` args over the `cluster use`?

That way there is no ambiguity in the commands, they are explicit and precise about what they select for deletion - if for example a user were able to fluke an accidental concurrent `confluent` command use in another shell, these deletes would remain correct and work even if the `cluster use` has been undone, or otherwise failed.

Reply:
I understand the goal but there are so many other places in this ccloud library (aside from this function) where concurrent `confluent` commands can result in problems, that IMO it's safer to not set that expectation with the user.  It could be a separate effort to harden those functions to meet that expectation.  For now, given what little cycles we have for `ccloud-stack` in general, IMO it's better to optimize for simpler code.